### PR TITLE
Apply various fixes given by clang-tidy

### PR DESCRIPTION
**Pull request by dundargoc**
Replace type of did_newline from int to bool.
Replace assert() with static_assert() if possible.
Use nullptr over NULL.
Remove redundant if-statement.
Remove redundant function declarations.
Convert standard for loops to range-based for loops.

### CODE REVIEW

**Comment by dundargoc** [comment]
This was mostly done to shut up my linter. I considered just adding the warnings to my ignore-list but thought it couldn't hurt to just send a PR to fix them :).

**Review by gmaurel — CHANGES_REQUESTED** [review]
Please activate the  workflows.

**Comment by dundargoc** [comment]
Done :)

File: src/uncrustify.cpp [hunk 1]
```diff
@@ -1583,9 +1583,9 @@ static void do_source_file(const char *filename_in,
 
    if (cpd.if_changed)
    {
-      for (deque<UINT8>::const_iterator i = cpd.bout->begin(), end = cpd.bout->end(); i != end; ++i)
+      for (unsigned char i : *cpd.bout)
```
Comment (micheleCTDE) [inline src/uncrustify.cpp:1586]: This should be UINT8 and not 'unsigned char'. Yes, they are likely to be the same underlining type, but it is cleaner to use the same type of the container.

**Review by micheleCTDE — CHANGES_REQUESTED** [review]
Thanks, look mostly good. Please see comments below.

File: src/output.cpp [hunk 2]
```diff
@@ -1770,8 +1770,7 @@ static Chunk *output_comment_c(Chunk *first)
       tmp.set(pc->str, 2, pc->Len() - 4);
 
       if (  cpd.last_char == '*'
-         && (  tmp[0] == '/'
```
[the remark this answers is not in the document]
Comment (dundargoc) [reply src/output.cpp:1773]: It doesn't really hurt, it's mostly that it's redundant. `tmp[0] != ' '` is always true if `tmp[0] == '/'` is true.

File: src/uncrustify.cpp [hunk 3]
```diff
@@ -427,14 +427,14 @@ int main(int argc, char *argv[])
    // output.cpp, line 427: fprintf(pfile, "# Line              Tag                Parent...
    // and              430: ... fprintf(pfile, "%s# %3zu>%19.19s[%19.19s] ...
    // here                                                xx xx   xx xx
-   for (size_t token = 0; token < ARRAY_SIZE(token_names); token++)
+   for (auto &token_name : token_names)
```
[the remark this answers is not in the document]
Comment (dundargoc) [reply src/uncrustify.cpp:430]: I think the const is already "baked in" into auto, but I don't think it hurts to be sure anyway :)

File: src/uncrustify.cpp (hunk 1 above)
Comment (dundargoc) [reply src/uncrustify.cpp]: Good point. Thanks.

File: src/output.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (micheleCTDE) [reply src/output.cpp:1773]: ups! it was a ```!= ' '```, not a ```== ' '```. Sorry, totally missed that, my bad. I read it as a ```==```. No issues then.

**Review by micheleCTDE — APPROVED** [review]
Looks good.

**Review by micheleCTDEAdmin — APPROVED** [review]

**Comment by micheleCTDEAdmin** [comment]
@gmaurel you approval is requested since you commented on the PR. I could force-merged, but I prefer to wait for your green tick

**Review by gmaurel — APPROVED** [review]

**Pull request merged**